### PR TITLE
Index files on first open

### PR DESCRIPTION
**Pull request by wz1000**
We don't want to index files on every change as this is needlessly expensive, but it is good to index them on first open so that information from the file is available in the next ghcide session.

### CODE REVIEW

File: ghcide/src/Development/IDE/LSP/Notifications.hs [hunk 1]
```diff
@@ -47,15 +47,15 @@ setHandlersNotifications = PartialHandlers $ \WithMessage{..} x -> return x
...
+                modifyFilesOfInterest ide (M.insert file Modified{firstOpen=True})
                 setFileModified ide False file
                 logInfo (ideLogger ide) $ "Opened text document: " <> getUri _uri
 
     ,LSP.didChangeTextDocumentNotificationHandler = withNotification (LSP.didChangeTextDocumentNotificationHandler x) $
         \_ ide (DidChangeTextDocumentParams identifier@VersionedTextDocumentIdentifier{_uri} changes) -> do
             updatePositionMapping ide identifier changes
             whenUriFile _uri $ \file -> do
-              modifyFilesOfInterest ide (M.insert file Modified)
+              modifyFilesOfInterest ide (M.insert file Modified{firstOpen=False})
```
Comment (Ailrun) [inline ghcide/src/Development/IDE/LSP/Notifications.hs:58]: Does this work when we rename a file?

Comment (wz1000) [reply ghcide/src/Development/IDE/LSP/Notifications.hs:58]: Why wouldn't it?

Comment (Ailrun) [reply ghcide/src/Development/IDE/LSP/Notifications.hs:58]: Hm... maybe I missed something. It actually works.

**Comment by pepeiborra** [comment]
Are there any downsides to not reindexing FOIs?

**Review by pepeiborra — APPROVED** [review]
worth adding a test?

**Comment by wz1000** [comment]
The most straightforward test I can think of will only show observable differences when https://github.com/haskell/haskell-language-server/pull/1357 lands. I'll add a test case after that.

BTW we already do index FOIs on save.

**Pull request merged**